toolchain/parse/testdata/function/declaration.carbon

```diff
@@ -36,6 +36,14 @@ fn foo(a: i32, b: i32);
 
 fn foo() -> u32;
 
+// --- keyword.carbon
+
+fn destroy() {}
+
+// --- keyword_no_params.carbon
+
+fn destroy {}
+
```

Comment:
Some other cases? like `fn MyClass.destroy()` or `fn destroy.Unexpected()`

Reply:
Added more tests; I'd missed in particular that `destroy.Unexpected()` makes a need for a IdentifierNameQualifierWithoutParams/KeywordNameQualifierWithoutParams split, so thanks for that catch.